webapp/sources/rudder/rudder-rest/src/main/scala/com/normation/rudder/rest/lift/RuleApi.scala

```diff
@@ -755,10 +755,11 @@ class RuleApiService14 (
 
   def getRuleApplicationStatus(rule: Rule, groupLib: FullNodeGroupCategory, directiveLib: FullActiveTechniqueCategory, nodesLib: Map[NodeId, NodeInfo], globalMode: GlobalPolicyMode): RuleApplicationStatus = {
     val directives = rule.directiveIds.flatMap(directiveLib.allDirectives.get(_)).map{case (a,d)=> (a.toActiveTechnique(), d)}
-    val nodes = groupLib.getNodeIds(rule.targets, nodesLib).flatMap(nodesLib.get)
+    val nodesIds = groupLib.getNodeIds(rule.targets, nodesLib)
+    val nodes = nodesLib.view.filterKeys( x => nodesIds.contains(x)).values
```

Comment:
impact of this is simply delirious. it goes from 9s to 6s with only this

Reply:
can you add a comment ? If it is significatif, it must be kept in futur refactoring

Reply:
a unit test would be more futur proof, but I have no idea how to build a relevant test for that, that won't fail randomly depending of the machine, and so being just noise

Reply:
i'm unsure also, because all other perfs tests are disabled :(